prober/tcp.go

```diff
@@ -159,6 +160,31 @@ func ProbeTCP(ctx context.Context, target string, module config.Module, registry
 			probeFailedDueToRegex.Set(0)
 			send = string(qr.Expect.Regexp.Expand(nil, []byte(send), scanner.Bytes(), match))
 		}
+		if qr.ExpectBytes != "" {
```

Comment:
This looks OK to me.

I would prefer to have stronger validation in the configuration, because as this section of the code shows, it's possible to specify both a regular expression and the sequence of bytes to be read, and I'm not sure how that's going to play together. If you think specifying both at the same time makes sense, I would like to see an example (and also a test).

My hunch is that it has to be either the regular expression or the specific sequence of bytes, but not both at the same time.

Reply:
It would be awesome if we get this MR merged and release :)

Reply:
I would also want `expect` and `expect_bytes` to be mutually exclusive.

> protocols are either line oriented or not. I'll adjust the patch soon.

@tacho is this done? I am happy to give it another review once the comments are addressed.